cli/server.go

```diff
@@ -875,9 +876,17 @@ func (r *RootCmd) Server(newAPI func(context.Context, *coderd.Options) (*coderd.
 			defer provisionerdWaitGroup.Wait()
 			provisionerdMetrics := provisionerd.NewMetrics(options.PrometheusRegistry)
 			for i := int64(0); i < vals.Provisioner.Daemons.Value(); i++ {
+				suffix := fmt.Sprintf("%d", i)
+				hostname := cliutil.Hostname()
+				// The suffix is added to the hostname, so we may need to trim to fit into
+				// the 64 character limit.
+				if len(hostname+suffix) > 62 {
+					hostname = hostname[:62-len(suffix)]
+				}
+				name := fmt.Sprintf("%s-%s", hostname, suffix)
```

Comment:
A hostname could include e.g. `_`, `/`, `--`, etc. I believe validation would fail in this case.. should we sanitize (strip/replace/allowlist)?

Reply:
It _should_ not, but it turns out that you can write whatever you want to `/proc/sys/kernel/hostname` but the `hostname` command will validate its input. It looks like `docker` will happily do this as well.

If we trim, we run the risk of collisions between machines named `foo/bar`, `foo~bar`, and `foo!"£$%^&*()_+{}~@:?,bar`. I'm leaning towards not sanitizing this.